images/access-setup/content/bin/common.sh

```diff
@@ -37,8 +37,13 @@ get_context() {
     printf "[ERROR] Install jq\n" >&2
     exit 1
   fi
-  token_secret="$(KUBECONFIG="$target" kubectl get sa "$sa" -n "$namespace" -o json |
-    jq -r '.secrets[].name | select(. | test(".*token.*"))')"
+
+  token_secret=""
+  while [ -z "$token_secret" ]; do
+    token_secret="$(KUBECONFIG="$target" kubectl get sa "$sa" -n "$namespace" -o json |
+    jq -r '.secrets[]?.name | select(. | test(".*token.*"))')"
+  done
```

Comment:
Shouldn't we have a timeout as well? In case, `token_secret` never gets populated.

Reply:
yes good point

Reply:
fixed

Reply:
there should be a sleep 1 between the trials. It is in my opinion better than just relying on the latency of the API server.

Reply:
discussed with @fgiloux ,  it should not take more than a couple of seconds and we should not let wait the user for more than 1 minute. we decided to be sleep 5 and loop 5 times. 
Fixed

Reply:
sorry Xin I was not really clear. The sleep 5s should only happen when it is needed. If the first call is successful we should not make the process wait.
I would propose to do a bit of refactoring, something like:
~~~
for i in {1..5}
do
    token_secret="$(KUBECONFIG="$target" kubectl get sa "$sa" -n "$namespace" -o json |
      jq -r '.secrets[]?.name | select(. | test(".*token.*"))')"
    if [ -n "$token_secret" ]; then
       break
    fi
    sleep 5
done
if [ -z "$token_secret" ]; then
    printf "Failed to extract secret token\n"
    exit 1
fi
~~~

Reply:
thank you for your suggestion, your logic look more clear. I directly copied your codes. Fixed.

Reply:
old logic needs to be trimmed

Reply:
done.